### PR TITLE
#20 test transform

### PR DESCRIPTION
**Pull request by SurinaCS**
Written tests to cover edge cases of transform (such as if there is no csv to extract keywords from) and amended transform.py to include error handling. Updated the readme to reflect content of test_transform.py. Achieves 97% pytest coverage.
Addresses issue #20

### CODE REVIEW

File: clean/clean.py [hunk 1]
```diff
@@ -24,9 +25,23 @@
...
+        s3 = client("s3", aws_access_key_id,
+                    aws_secret_access_key)
+        return s3
+    except (NoCredentialsError, PartialCredentialsError) as e:
+        logging.error("A BotoCore error occurred: %s", e)
+        raise
+    except Exception as e:
+        logging.error(
+            "An unexpected error occurred while connecting to S3: %s", e)
+        raise
```
Comment (keoghrmj) [inline clean/clean.py:44]: Lovely error handling

File: pipeline/transform.py [hunk 2]
```diff
@@ -74,9 +81,19 @@ def keyword_matching(cleaned_bluesky_data: pd.DataFrame, keyword_map: dict) -> p
...
+            return bluesky_data['Keyword'].unique()
+
+        raise FileNotFoundError(f"File {csv_file} does not exist")
+    except FileNotFoundError as e:
+        logging.error('File not found at path %s: %s', csv_file, e)
+        raise
+    except Exception as e:
+        logging.error(
+            "An error occurred while reading the file %s: %s", csv_file, e)
+        raise
```
Comment (keoghrmj) [inline pipeline/transform.py:96]: Again, lovely error handling!

**Review by keoghrmj — APPROVED** [review]
Really nice tests and error handling
